src/common/thumbnail-manager.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+import {Utils, ThumbnailInfo, MediaType} from '@playkit-js/playkit-js';
+import evaluate from './utils/evaluate';
+
+const DefaultThumbnailConfig: Object = {
+  thumbsWidth: 164,
+  thumbsHeight: 92,
+  thumbsSlices: 100
+};
+
+const THUMBNAIL_REGEX = /.*\/p\/\d+\/(?:[a-zA-Z]+\/\d+\/)*thumbnail\/entry_id\/\w+\/.*\d+/;
+const THUMBNAIL_SERVICE_TEMPLATE: string = '{{thumbnailUrl}}/width/{{thumbsWidth}}/height/{{thumbsHeight}}/vid_slices/{{thumbsSlices}}/ks/{{ks}}';
+
+class ThumbnailManager {
+  _player: Player;
+  _thumbnailConfig: ?SeekbarConfig;
+
+  constructor(mediaConfig: KPMediaConfig, player: Player) {
+    this._player = player;
+    this._thumbnailConfig = this._buildKalturaThumbnailConfig(mediaConfig);
```

Comment:
Maybe I don't remember but why wouldn't we create track? I saw that you actually can request from thumbnail service multiple different parameters so you can provide parameters that will change the track on every get thumbnail request with different parameters and then it reasonable(same we do for external track) not sure if it's needed to expose all the options but I think we take it into considartion.